index.js

```diff
@@ -1,37 +1,129 @@
-const hash = CryptoJS.MD5(ts + privateKey + publicKey).toString();
+const date = Date.now().toString();
+const hash = CryptoJS.MD5(date + privateKey + publicKey).toString();
 
-const url = `https://gateway.marvel.com/v1/public/characters?ts=${ts}&apikey=${publicKey}&hash=${hash}`;
+const images = [
+    'image1.jpg',
+    'image2.jpg',
+    'image3.jpg',
+];
 
-fetch(url)
-    .then(function (response) {
-        return response.json();
-    })
+document.addEventListener('DOMContentLoaded', function() {
+    setInterval(changeBackground, 4000);
+    changeBackground();
+});
+
+function changeBackground() {
+    const randomImage = images[Math.floor(Math.random() * images.length)];
+    document.body.style.backgroundImage = `url('${randomImage}')`;
+}
+
+function handleMainPage() {
+    setInterval(changeBackground, 4000);
+    changeBackground();
 
-    .then(function (data) {
-        console.log(data);
+    document.getElementById('searchForm').addEventListener('submit', function (event) {
+        event.preventDefault();
 
+        const characterName = document.getElementById('searchInput').value;
 
-    })
+        if (characterName) {
+            const url = `https://gateway.marvel.com/v1/public/characters?nameStartsWith=${characterName}&ts=${date}&apikey=${publicKey}&hash=${hash}`;
 
-    .catch(function (error) {
-        console.error('Error fetching data:', error);
+            fetch(url)
+                .then(response => response.json())
+                .then(data => {
+                    console.log(data);
+                    const hero = data.data.results[0];
+
+                    if (hero) {
+                        document.getElementById('heroInfo').innerHTML = `
+                            <img src="${hero.thumbnail.path}.${hero.thumbnail.extension}" alt="${hero.name}">
+                            <h2>${hero.name}</h2>
+                            <p>${hero.description || 'No description available.'}</p>
+                        `;
+                    } else {
+                        document.getElementById('heroInfo').innerHTML = '<p>Character not found.</p>';
+                    }
+                })
+                .catch(error => {
+                    console.error('Error fetching data:', error);
+                    document.getElementById('heroInfo').innerHTML = '<p>Error retrieving data.</p>';
+                });
+        }
     });
 
-    const images = [
-        'image1.jpg',
-        'image2.jpg',
-        'image3.jpg',
-    ]
+    document.getElementById('heroInfo').insertAdjacentHTML('afterend', `
+        <nav>
+            <button id="comicsNavButton">Go to Comics List</button>
+        </nav>`);
 
-    function changeBackground() {
-        const randomImage = images[Math.floor(Math.random() * images.length)];
-        document.body.style.backgroundImage = `url('${randomImage}')`;
-    }
+    document.getElementById('comicsNavButton').addEventListener('click', function () {
```

Comment:
Instead of adding a dynamic button to go to the comics page. Use a regular `a` element to navigate to the comics page. Then, you can have a separate `comics.js` file imported on the other page, that might make it simpler for you to handle displaying different data for each page.